### ISSUE TITLE
useAppCheck and/or preloadAppCheck

### ISSUE THREAD

**Issue by MaciejCaputa** [FirebaseExtended/reactfire#406]
Hi, I think it would be a good idea to add `useAppCheck` and/or `preloadAppCheck` to provide functionality form https://firebase.google.com/docs/app-check

### Version info

**React: ^17.0.2**

**Firebase: ^8.7.1**

**ReactFire: ^3.0.0-rc.2**

### Steps to reproduce

1. Add `import 'firebase/app-check'`;
2. Add `useFirebaseApp().appCheck().activate('YOUR KEY')`

### Expected behavior
I have an app check working.

### Actual behavior

```
Uncaught FirebaseError: AppCheck: You are trying to activate AppCheck for FirebaseApp [DEFAULT], while it is already activated. AppCheck can only be activated once. (appCheck/already-activated).
```

**Comment by epodol** [comment]
I'd be happy to try my hand at adding support for App Check, but I am not sure if @jhuleatt wants to wait until #369 is completed. 

If your looking for a temporary solution, here is how I am using it:

```javascript
import 'firebase/app-check';
// ...
  const firebaseApp = useFirebaseApp();

  const appCheck = !isDev ? firebaseApp.appCheck() : null;

  useEffect(() => {
    if (!isDev && process.env.REACT_APP_RECAPTCHA_PUBLIC_KEY)
      appCheck?.activate(process.env.REACT_APP_RECAPTCHA_PUBLIC_KEY);
  }, [appCheck]);
  ```
https://github.com/epodol/bsclibrary/blob/2a5d1cd3494d42859a21436c783daa1db649da1d/react/src/App.tsx#L114

**Comment by jhuleatt** [comment]
Thanks @MaciejCaputa for the feature request, and thank you @epodol for your temporary solution!

This is definitely an important feature to add to ReactFire. Let's discuss a potential API. I wonder if we should have another Provider for AppCheck. It seems important to be able to be confident that AppCheck is activated before any calls are made that could be denied by AppCheck.

For example, it could look like:

```jsx
<FirebaseAppProvider firebaseConfig={/* config... */}>
    <AppCheckProvider publicKey={/* recaptcha public key... */}>
        <MyChatApp />
    </AppCheckProvider>
</FirebaseAppProvider>
```

Then, I could be confident that any Realtime Database, Cloud Storage, or Cloud Functions (callable functions) calls that I make within `<MyChatApp />` are made _after_ AppCheck is activated.

This could also be done in `FirebaseAppProvider`, but separating it out may be good to reduce bundle size for those that aren't using AppCheck.

The props for `AppCheckProvider` would likely have to account for [all of the arguments `activate` accepts](https://firebase.google.com/docs/reference/js/firebase.appcheck.AppCheck#activate): `siteKeyOrProvider :  string | AppCheckProvider ,  isTokenAutoRefreshEnabled ? : boolean`

---

How does this sound to you? How can we make it better?

**Comment by jhuleatt** [comment]
If we don't want the component props to match `activate`'s api, maybe it could be a single callback prop instead:

```jsx
<AppCheckProvider
    activateCallback={(activate) => activate(process.env.REACT_APP_RECAPTCHA_PUBLIC_KEY)}
>
```

**Comment by epodol** [comment]
A provider for App Check would probably be ideal. In my opinion, an `activateCallback` would be better because it would allow for more customization for individual use cases, and reduce any breaking api changes in the future if firebase changes the activate arguments.

**Comment by MaciejCaputa** [comment]
I think it's a good idea. :)

**Comment by jhuleatt** [comment]
App Check is available in ReactFire [`4.1.0`](https://github.com/FirebaseExtended/reactfire/releases/tag/v4.1.0)!

Take a look at the [Using ReactFire guide](https://github.com/FirebaseExtended/reactfire/blob/main/docs/use.md#set-up-app-check) for a code sample.
